src/tests/editRequest.test.js

```diff
@@ -0,0 +1,184 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import mockData from './mockData';
+import app from '../app';
+import isObjectEmpty from '../utils/isObjectEmpty';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+let requestIdToEdit;
+const testEditRequest = () => {
+  describe('User can edit open request', () => {
+    it('should return 200 on successful signIn', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/auth/login')
+        .send({
+          email: 'knights@gmail.com',
+          password: 'Niyonkuru@1'
+        })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(200);
+          done();
+        });
+    });
+    it('should return 422 if user has no manager ', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/trips/returnTrip')
+        .send({
+          origin: 'Kigali',
+          destination: 'Kampala',
+          departureDate: '2023-01-11',
+          returnDate: '2023-01-15',
+          reason: 'Having fun',
+          accommodation: 'Z campus'
+        })
+        .end((err, res) => {
+          expect(res.status).to.equal(422);
+          expect(res.body).to.have.property('error').that.equals('you currently have no lineManager, please go to update your profile');
+          done();
+        });
+    });
+    it('should return 200 on successful signIn', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/auth/login')
+        .send({
+          email: 'multi-city@gmail.com',
+          password: 'Niyonkuru@1'
+        })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(200);
+          done();
+        });
+    });
+    it('should return 200 on successful created request ', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/trips/returnTrip')
+        .send({
+          origin: 'Kigali',
+          destination: 'Kampala',
+          departureDate: '2023-01-11',
+          returnDate: '2023-01-15',
```

Comment:
This is too much information need to be in separate file

Reply:
I love them inside because of the clearer it makes on what is being tested.